fdbserver/DataDistribution.actor.cpp

```diff
@@ -2569,6 +2577,196 @@ ACTOR Future<Void> waitUntilHealthy(DDTeamCollection* self, double extraDelay =
 	}
 }
 
+// Take a snapshot of necessary data structures from `DDTeamCollection` and print them out with yields to avoid slow
+// task on the run loop.
+ACTOR Future<Void> printSnapshotTeamsInfo(Reference<DDTeamCollection> self) {
+	state DatabaseConfiguration configuration;
+	state std::map<UID, Reference<TCServerInfo>> server_info;
+	state std::map<UID, ServerStatus> server_status;
+	state vector<Reference<TCTeamInfo>> teams;
+	state std::map<Standalone<StringRef>, Reference<TCMachineInfo>> machine_info;
+	state std::vector<Reference<TCMachineTeamInfo>> machineTeams;
+	// state std::vector<std::string> internedLocalityRecordKeyNameStrings;
+	// state int machineLocalityMapEntryArraySize;
+	// state std::vector<Reference<LocalityRecord>> machineLocalityMapRecordArray;
+	state int traceEventsPrinted = 0;
+	state std::vector<const UID*> serverIDs;
+	state double lastPrintTime;
+	loop {
+		wait(self->printDetailedTeamsInfo.onTrigger());
+		if (now() - lastPrintTime < SERVER_KNOBS->DD_TEAMS_INFO_PRINT_INTERVAL) {
+			continue;
+		}
+		lastPrintTime = now();
+
+		traceEventsPrinted = 0;
+
+		double snapshotStart = now();
+
+		configuration = self->configuration;
+		server_info = self->server_info;
+		teams = self->teams;
+		machine_info = self->machine_info;
+		machineTeams = self->machineTeams;
+		// internedLocalityRecordKeyNameStrings = self->machineLocalityMap._keymap->_lookuparray;
+		// machineLocalityMapEntryArraySize = self->machineLocalityMap.size();
+		// machineLocalityMapRecordArray = self->machineLocalityMap.getRecordArray();
+		std::vector<const UID*> _uids = self->machineLocalityMap.getObjects();
+		serverIDs = _uids;
+
+		auto const& keys = self->server_status.getKeys();
+		for (auto const& key : keys) {
+			server_status.emplace(key, self->server_status.get(key));
+		}
+
+		TraceEvent("DDPrintSnapshotTeasmInfo", self->distributorId)
+		    .detail("SnapshotSpeed", now() - snapshotStart)
+		    .detail("Primary", self->primary);
```

Comment:
I should have written the convention in the previous review comment:
we typically put "Primary" field as the first field of the event. That makes it easier to tell which DC the event belong to. 

I won't insist on changing this, though.

Reply:
IMO the positioning is really not important, since most of the time those events are being filtered.